src/Routing/Router.php

```diff
@@ -924,4 +929,29 @@ public static function setRouteCollection(RouteCollection $routeCollection): voi
     {
         static::$_collection = $routeCollection;
     }
+
+    /**
+     * Inject route defaults from `_path` key
+     *
+     * @param array $url Route array with `_path` key
+     * @return array
+     */
+    protected static function unwrapShortString(array $url)
```

Comment:
Should we cache the results of this method into an in-memory array that is cleared by `Router::reload()`? I can see this method getting hit quite a few times in a request and doing redundant work.

Reply:
For each `Router::url()` call with `_path` key, parsing string will be done once. Further route matching at collection takes much more work.

Reply:
What about pages that have 20+ links on them? We'll be going through this block of code 20+ times and we could pretty easily cache the results.